core/finality-grandpa/src/communication/gossip.rs

```diff
@@ -717,41 +721,40 @@ impl<Block: BlockT> network_gossip::Validator<Block> for GossipValidator<Block>
 		};
 
 		Box::new(move |who, intent, topic, mut data| {
-			if let MessageIntent::PeriodicRebroadcast = intent {
-				return do_rebroadcast;
-			}
-
-			let peer = match inner.peers.peer(who) {
-				None => return false,
-				Some(x) => x,
-			};
-
-			// if the topic is not something we're keeping at the moment,
-			// do not send.
-			let (maybe_round, set_id) = match inner.live_topics.topic_info(&topic) {
-				None => return false,
-				Some(x) => x,
-			};
-
-			// if the topic is not something the peer accepts, discard.
-			if let Some(round) = maybe_round {
-				return peer.view.consider_vote(round, set_id) == Consider::Accept
-			}
-
-			// global message.
-			let our_best_commit = inner.local_view.last_commit;
-			let peer_best_commit = peer.view.last_commit;
-
+			
 			match GossipMessage::<Block>::decode(&mut data) {
 				None => false,
 				Some(GossipMessage::Commit(full)) => {
-					// we only broadcast our best commit and only if it's
-					// better than last received by peer.
-					Some(full.message.target_number) == our_best_commit
-					&& Some(full.message.target_number) > peer_best_commit
+					if let MessageIntent::PeriodicRebroadcast = intent {
+						return do_rebroadcast;
+					}
+					Some(full.message.target_number) >= inner.local_view.last_commit
```

Comment:
The previous behavior of this function was to only broadcast our best commit. The new behavior is to broadcast any commit greater than our best. There are a couple problems:
  1. We may send a peer a commit which is `<` their best (there used to be a check). This is impolite
  2. We may send a peer multiple commits, but not in ascending order. e.g. we send `a` and `b` but `b < a`. This is impolite.
  3. We may send commits which haven't been fully validated. This is the reason we only send our best commit. We may be aware of commit messages which _claim_ to target a higher block, but we have the `CommitProcessingOutcome` callback that tells us whether to repropagate for real.